AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -527,16 +538,8 @@ - (void)accessDataSourceSynchronously:(BOOL)synchronously withBlock:(dispatch_bl
       NSIndexPath *indexPath = [sectionIndex indexPathByAddingIndex:i];
       ASCellNodeBlock nodeBlock = [_dataSource dataController:self nodeBlockAtIndexPath:indexPath];
       
-      // When creating a node, make sure to pass along the current display traits so it will be laid out properly
-      ASCellNodeBlock nodeBlockPropagatingDisplayTraits = ^{
-        ASCellNode *cellNode = nodeBlock();
-        id<ASEnvironment> environment = [self.environmentDelegate dataControllerEnvironment];
-        ASEnvironmentStatePropagateDown(cellNode, [environment environmentTraitCollection]);
-        return cellNode;
-      };
-      
       ASSizeRange constrainedSize = [self constrainedSizeForNodeOfKind:ASDataControllerRowNodeKind atIndexPath:indexPath];
-      [contexts addObject:[[ASIndexedNodeContext alloc] initWithNodeBlock:nodeBlockPropagatingDisplayTraits
+      [contexts addObject:[[ASIndexedNodeContext alloc] initWithNodeBlock:[self nodeBlockPropagatingTraitCollectionFromNodeBlock:nodeBlock]
```

Comment:
These codepaths are extremely hot - thus nasty stuff like manually malloc'ing memory and creating NSArrays :).  We should not wrap an additional block as it adds retain / release overhead (I believe it will not only retain the block, but also everything referenced inside it - not certain about the latter piece).

It seems like we could instead implement the propogation step at the point where the existing nodeBlock is called rather than swapping out the block implementation.